### ISSUE TITLE
Strategies for creating localized Yasgui 

### ISSUE THREAD

**Issue by EmmanuelDemey** [TriplyDB/Yasgui#177]
Hi, 

For one of my client, we need to use Yasgui but with all texts translated into french. 

We will be very happy to contribute to your project and provide all the french translations. 

But if I am right, for the moment, nothing is available in the project in order to make this task easy :D 

Before starting contributing to the project, do you have any ideas about making Yasgui localized ? 

Regards

**Comment by LaurensRietveld** [comment]
There are some issues with localization. Yasgui relies on some third-party UI-heavy libraries such as codemirror, or for plugins such as geo, geo3d, charts, pivot, geo-events and timeline. These libaries may or may not support localized labels. 
Because of these dependencies that are difficult to localize, we've decided to keep localization support a low priority as we can only deliver a partial localized experience.

I'm open to any suggestions though if you have a proposal on how to solve this localization issue across Yasgui and it's dependencies!

**Comment by bzar** [comment]
Bumped into this as well.

Codemirror [supports translations with a simple phrase mapping](https://codemirror.net/examples/translate/). DataTables [supports translations with a similar system](https://datatables.net/manual/i18n). I wonder if Yasgui could piggy-back on these and support providing a single phrases object, which could be used as both CM and DT configuration and internally with Yasgui specific text? I mean Yasgui could have a single configuration option for setting a translation object, which (or parts of which) would be passed to the dependencies?

**Comment by tfrancart** [comment]
Would like to see this happening as well. At the very least I would need to localize the YasR tab names.
